src/smtp-lib/mail_store.cpp

```diff
@@ -1,51 +1,55 @@
 #include "mail_store.hpp"
 
-#include <algorithm>
-#include <fstream>
-#include <iostream>
-#include <random>
-#include <string>
-
-#include "mail.hpp"
-
-#if __has_include(<filesystem>)
-#include <filesystem>
-namespace fs = std::filesystem;
-#else
-#include <experimental/filesystem>
-namespace fs = std::experimental::filesystem;
-#endif
+#include <ctime>
+
+#include <cpr/cpr.h>
+#include <nlohmann/json.hpp>
 
 namespace {
-static const fs::path MAIL_FOLDER = "mails";
-
-std::string random_string(const size_t length) {
-    auto randchar = []() {
-        static const char charset[] =
-            "0123456789"
-            "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
-            "abcdefghijklmnopqrstuvwxyz";
-        const size_t max_index = (sizeof(charset) - 1);
-        return charset[rand() % max_index];
-    };
-    std::string str(length, 0);
-    std::generate_n(str.begin(), length, randchar);
-    return str;
+    void fix_time_value(std::string& time_value) {
+        if (time_value.length() == 1) {
+            time_value.insert(time_value.begin(), '0');
+        }
+    }
+
+    std::string time_string() {
+        const auto elapsed_time = std::time(nullptr);
+        const auto utc_time = std::gmtime(&elapsed_time);
+
+        const auto year = std::to_string(utc_time->tm_year + 1900);
+        auto month = std::to_string(utc_time->tm_mon + 1);
+        auto day = std::to_string(utc_time->tm_mday);
+        auto hour = std::to_string(utc_time->tm_hour);
+        auto minute = std::to_string(utc_time->tm_min);
+        auto second = std::to_string(utc_time->tm_sec);
+
+        fix_time_value(month);
+        fix_time_value(day);
+        fix_time_value(hour);
+        fix_time_value(minute);
+        fix_time_value(second);
+
+        return std::string{year + '-' + month + '-' + day + 'T' + hour + ':' + minute + ':' + second + ".00+00:00"};
+    }
 }
-}  // namespace
 
 bool MailStore::store_mail(const Mail& mail) {
-    fs::create_directory(MAIL_FOLDER);
-    const auto str = random_string(6);
-    std::cout << str << std::endl;
-    fs::path mail_path = MAIL_FOLDER / mail.from / "-" / str;
-    std::ofstream mail_file{mail_path};
-    mail_file << "FROM: " << mail.from << std::endl;
-
-    for (const auto& recipient : mail.to) {
-        mail_file << "TO: " << recipient << std::endl;
+    nlohmann::json json;
+    json["received"] = time_string();
+    json["received_from"] = "sending.server";
+    json["received_by"] = "receiving.server";
+    json["mail_from"] = mail.from;
+    json["rcpt_to"] = mail.to;
+    json["data"] = mail.data;
+
+    for (auto i{0}; i < 5; ++i) {
+        auto response =
+            cpr::Post(cpr::Url{"http://localhost:5000/mailstore"}, cpr::Header{{"Content-Type", "application/json"}},
+                      cpr::Body{json.dump(4)}, cpr::Timeout{5000});
```

Comment:
what does the `dump(4)` do?

Reply:
That's the indent used for formatting the string